Monika After Story/game/script-topics.rpy

```diff
@@ -1854,6 +1842,295 @@ label monika_rain_holdme:
+            m 2hubfa "Mmm~"
+            m 1ekbfb "That was really nice, [player]."
+            m 1hubfb "Long hugs are supposed to wash away any stress."
+            m 1tubfb "Even if you weren't stressed, I hope you're feeling better after that."
+            m 1hubfa "I know I sure am~"
+            m 1hubfb "Ahaha!"
+        elif datetime.datetime.now() > (start_time + datetime.timedelta(seconds=30)):
+            m 1hubfa "Mmm~"
+            m 1hubfb "I can almost feel your warmth, even from here."
+            m 1eua "I'm sure you know hugs are good for you, since they relieve stress and all."
+            m 3eub "But did you know the sweet spot for hugs is 30 seconds?"
+            m 3hubfb "Hugs are most effective at relieving stress and being good for you when they last 30 seconds, ahaha!"
+            m 1eud "Oh, wait, did I say 30 seconds?"
+            show monika 5eubfu at t11 zorder MAS_MONIKA_Z with dissolve
+            m 5eubfu "Sorry, I meant {i}at least{/i} 30 seconds, ehehe..."
+            show monika 1hubfb at t11 zorder MAS_MONIKA_Z with dissolve
+        else:
+            #Under 30 seconds
+            m 1ekc "Aw, done holding me already, [player]?"
+            m 1rksdla "I was kind of hoping for it to last a little bit longer..."
+            m 1hubfa "I'm sure that won't be the last time you hold me though, so I'll look forward to next time!"
+    #happy
+    else:
+        if datetime.datetime.now() > (start_time + datetime.timedelta(minutes=30)):
+            call monika_holdme_long
+        elif datetime.datetime.now() > (start_time + datetime.timedelta(minutes=10)):
+            m 6dubsa "Hm?"
+            m 1wud "Oh! We're done?"
+            m 3hksdlb "That hug sure lasted a while, [player]..."
+            m 3rubfb "Nothing wrong with that, I just thought you'd let go a lot sooner, ahaha!"
+            m 1rkbsa "It was almost comfortable actually..."
+            m 2tubfb "Too much longer and I might have fallen asleep in your arms..."
+            m 1tubfa "Maybe you'd like that, ehehe~"
+            m 1hubfa "I feel all warm and nice now from that~"
+        elif datetime.datetime.now() > (start_time + datetime.timedelta(minutes=2)):
+            m 1hksdlb "That lasted a little while."
+            m 1rkbsa "Don't get me wrong- I still enjoyed it."
+            m 1tubfb "I hope you're satisfied~"
+            m 1hubfa "I'm happy with just sitting with you now."
+        elif datetime.datetime.now() > (start_time + datetime.timedelta(seconds=30)):
+            m 1hubfa "Ehehe~"
+            m 3eub "Did you enjoy that?"
+            m 1hua "I sure hope so~"
+            m 1hubfb "Hugs are supposed to be good for you after all."
+        else:
+            #Under 30 seconds
+            m 1hua "That was nice~"
+            m 1hubfb "You can hold me for longer if you feel like it, you know."
+    return
+
+label monika_holdme_long:
+    if mas_isMoniLove():
```

Comment:
instead of 4 menus, just do 1 menu with aff branches inside the menu options. Less repeated code.